lib/ex_unit/test/ex_unit/assertions_test.exs

```diff
@@ -336,6 +336,32 @@ defmodule ExUnit.AssertionsTest do
       "Expected exception SyntaxError but got FunctionClauseError (no function clause matching in :lists.flatten/1)" = error.message
   end
 
+  test "assert raise comparing messages (for equality)" do
+    assert_raise RuntimeError, "foo", fn ->
+      raise RuntimeError, "bar"
+    end
+  rescue
+    error in [ExUnit.AssertionError] ->
+      "Wrong message for RuntimeError" <>
+      "\nExpected:" <>
+      "\n  \"foo\"" <>
+      "\nGot:" <>
+      "\n  \"bar\"" = error.message
+  end
+
+  test "assert raise comparing messages (with a regex)" do
+    assert_raise RuntimeError, ~r/ba[zk]/, fn ->
+      raise RuntimeError, "bar"
+    end
+  rescue
+    error in [ExUnit.AssertionError] ->
+      "Wrong message for RuntimeError" <>
+      "\nExpected:" <>
+      "\n  ~r/ba[zk]/" <>
+      "\nGot:" <>
+      "\n  \"bar\"" = error.message
```

Comment:
@eksperimental honestly, I would wait to do that. The reason is: I always get annoyed by heredocs having a trailing newline, and would love a way to avoid that. For example, a modifier for the `~s`/`~S` sigils like `n` (for *n*o*n*newlines) so that you can write:

```elixir
~s"""
this heredoc has no trailing newlines
"""n
```

I'll probably write that on the core mailing list, is it ok for you to wait a little while before making these changes?